staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -439,3 +441,15 @@ func interpretPatchError(err error) error {
 		return err
 	}
 }
+
+// interpretJSONPatchError interprets the error and returns an error with appropriate HTTP code.
+func interpretJSONPatchError(err error) error {
+	switch {
+	case strings.Contains(err.Error(), "doc is missing"):
+		return errors.NewBadRequest(err.Error())
+	case strings.Contains(err.Error(), "Testing value") && strings.Contains(err.Error(), "failed"):
```

Comment:
I assume the library does not return specific error types for this?

Reply:
> I assume the library does not return specific error types for this?

Unfortunately, no :(

Reply:
I'm not in favor of string text matching for error recognition